### PR TITLE
update GfmFencesBlockRule.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ public abstract class Matcher
 
         private static readonly AnyCharMatcher AnyCharMatcher = new AnyCharMatcher();
         private static readonly EndOfStringMatcher EndOfStringMatcher = new EndOfStringMatcher();
+        private static readonly Matcher WhiteSpacesMatcher = Repeat(new CharMatcher(' '), 0);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs]: WhiteSpacesOrEmpty

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs [hunk 2]
```diff
@@ -63,9 +65,13 @@ public static Matcher AnyCharNotIn(params char[] ch)
             }
             var array = (char[])ch.Clone();
             Array.Sort(array);
-            return new AnyCharNotInMatcher(ch);
+            return new AnyCharNotInMatcher(array);
         }
 
+        public static Matcher WhiteSpaces => WhiteSpacesMatcher;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs]: can use get property

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs [hunk 3]
```diff
@@ -92,7 +93,7 @@ public static Matcher CaseInsensitiveString(string text)
             return new CaseInsensitiveStringMatcher(text);
         }
 
-        public static Matcher EndOfString() => EndOfStringMatcher;
+        public static Matcher EndOfString { get; } = new EndOfStringMatcher();
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs]: >EndOfString [](start = 30, length = 11)

minor: group the property together?

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**